workers/unity/Assets/Gdk/Core/Components/Option.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+
+namespace Improbable.Gdk.Core
+{
+    public struct Option<T> : IEquatable<Option<T>>
+    {
+        private bool hasValue;
+        private T value;
+        
+        public Option(T newValue)
+        {
+            hasValue = false;
+            value = default(T);
+            Value = newValue;
+        }
+
+        public BlittableBool HasValue => hasValue;
+
+        public T Value
+        {
+            get
+            {
+                if (!hasValue)
+                {
+                    throw new InvalidOperationException("Called Value on empty Option.");
```

Comment:
Please make a new exception type for this.

Reply:
Is this really necessary? What's wrong with the standard exceptions?

Reply:
Standard exceptions don't tell you what has gone wrong. You can find the message but the main thing you get is the exception type.

Reply:
Added bespoke exception type.